docs/how-to/hip_runtime_api/call_stack.rst

```diff
@@ -0,0 +1,132 @@
+by ensuring sufficient stack memory is allocated.
+
+.. code-block:: cpp
+        
+    #include <hip/hip_runtime.h>
+    #include <iostream>
+
+    #define HIP_CHECK(expression)                \
+    {                                            \
+        const hipError_t status = expression;    \
+        if(status != hipSuccess){                \
+                std::cerr << "HIP error "        \
+                    << status << ": "            \
+                    << hipGetErrorString(status) \
+                    << " at " << __FILE__ << ":" \
+                    << __LINE__ << std::endl;    \
+        }                                        \
+    }
+
+    int main()
+    {
+        size_t stackSize;
+        HIP_CHECK(hipDeviceGetLimit(&stackSize, hipLimitStackSize));
+        std::cout << "Default stack size: " << stackSize << " bytes" << std::endl;
+
+        // Set a new stack size
+        size_t newStackSize = 1024 * 8; // 1MB
+        HIP_CHECK(hipDeviceSetLimit(hipLimitStackSize, newStackSize));
+        std::cout << "New stack size set to: " << newStackSize << " bytes" << std::endl;
+
+        HIP_CHECK(hipDeviceGetLimit(&stackSize, hipLimitStackSize));
+        std::cout << "Updated stack size: " << stackSize << " bytes" << std::endl;
+
+        return 0;
+    }
+
+Handling recursion and deep function calls
+--------------------------------------------------------------------------------
+
+Similar to CPU programming, recursive functions and deeply nested function calls
+are supported. However, developers must ensure that these functions do not
+exceed the available stack memory, considering the limited resources on GPUs.
+
+.. code-block:: cpp
+
+    #include <hip/hip_runtime.h>
+    #include <iostream>
+    
+    #define HIP_CHECK(expression)                \
+    {                                            \
+        const hipError_t status = expression;    \
+        if(status != hipSuccess){                \
+                std::cerr << "HIP error "        \
+                    << status << ": "            \
+                    << hipGetErrorString(status) \
+                    << " at " << __FILE__ << ":" \
+                    << __LINE__ << std::endl;    \
+        }                                        \
+    }
+    
+    __device__ unsigned long long factorial(unsigned long long n)
+    {
+        if (n == 0 || n == 1) {
+            return 1;
+        }
+        return n * factorial(n - 1);
+    }
+    
+    __global__ void kernel(unsigned long long n)
+    {
+        unsigned long long result = factorial(n);
+        const size_t x = threadIdx.x + blockDim.x * blockIdx.x;
+        
+        if( x == 0)
+          printf("%llu! = %llu \n", n, result);
+    }
+    
+    int main()
+    {
+        kernel<<<1, 1>>>(10);
+        HIP_CHECK(hipDeviceSynchronize());
+
+        // With -O0 optimization option hit the stack limit
+        // kernel<<<1, 256>>>(2048);
```

Comment:
Also hits the stack limit with optimization, and is independent of block size, so it should use the same config as the first call
```suggestion
        // This hits the stack limit due to excessive recursion
        // kernel<<<1, 1>>>(2048);
```

Reply:
Thanks! Updated.

Reply:
I tested again on a different GPU, seems like this is somewhat unpredictable depending on the architecture, however with -O0 as you proposed earlier it "successfully" fails even on different architectures. Sorry, that was my bad, should have checked before with different GPUs.

Reply:
Reversed. Kept the original one.